app/src/main/java/com/github/gogiffy/activities/SearchActivity.java

```diff
@@ -50,15 +50,29 @@ public void beforeTextChanged(CharSequence s, int start, int count, int after) {
 
     @Override
     public void onTextChanged(CharSequence s, int start, int before, int count) {
-        if (timer != null) {
-            timer.cancel();
+        String invalid_variable_stye_here_its_too_long_too =    "" +
+                "adfafaf asfas fsafasff";
+        if (timer != null)     {
+
+                timer.cancel();
         }
+
+        //invalid_variable_stye_here_its_too_long_too.isEmpty()
+        /// Some other bad coding style
+
+          int x=4;
+        boolean isX=x==4;
     }
 
+    public void unUsedMethod(String x, String y){}
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS1186)